src/eth/rpc/rpc_server.rs

```diff
@@ -258,7 +261,12 @@ async fn stratus_health(_: Params<'_>, context: Arc<RpcContext>, _: Extensions)
     let should_serve = match GlobalState::get_node_mode() {
         NodeMode::Leader => true,
         NodeMode::Follower => {
-            let consensus_lock = context.consensus.read().map_err(|_| StratusError::ConsensusLockFailed)?;
+            let consensus_lock = context.consensus.read().map_err(|poison| {
+                tracing::error!("consensus read lock was poisoned");
+                context.consensus.clear_poison();
+                drop(poison.into_inner());
```

Comment:
nitpick: These drops are redundant as a local variable always drops by the end of the function (before the function returns any value)

Reply:
I receive warning if I do not use drop.
```
unused `std::sync::RwLockReadGuard` that must be used
if unused the RwLock will immediately unlock
`#[warn(unused_must_use)]` on by defaultrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fgabrielmartins%2FCloudWalk%2Fstratus%2Fsrc%2Feth%2Fminer%2Fminer.rs)
miner.rs(92, 17): use `let _ = ...` to ignore the resulting value: `let _ = `
```

I figured using drop is cleaner than using the `let _ = ...`

Reply:
Is something like this equivalent?
```
let consensus_lock = context.consensus.read().map_err(|_| {
                tracing::error!("consensus read lock was poisoned");
                context.consensus.clear_poison();
```